### PR TITLE
Refactor: move getAccount to accounts package (from openrtb2)

### PR DESCRIPTION
**Pull request by laurb9**
Move GetAccount() to a new accounts package, so it can be invoked from other modules.
Leaving the actual config model in config, to avoid changing too much.

### CODE REVIEW

**Review by SyntaxNode — COMMENTED** [review]
Nice. I like this. Just one nitpick. The preferred naming convention for packages in Go is to use a singular form. In this case, that would be `account` vs `accounts`. I understand this project already deviates quite a bit and this is more of a guideline than a rule. 

In usage, `account.GetAccount` looks a little nicer to me than `accounts.GetAccount`, but just barely.

Thoughts?

**Comment by laurb9** [comment]
> The preferred naming convention for packages in Go is to use a singular form. In this case, that would be `account` vs `accounts`.

Agree, will change it. Almost did it the first time around actually, then I saw all the other package names :)

File: account/account.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	if cfg.AccountRequired && accountID == pbsmetrics.PublisherUnknown {
+		return nil, []error{&errortypes.AcctRequired{
+			Message: fmt.Sprintf("Prebid-server has been configured to discard requests without a valid Account ID. Please reach out to the prebid server host."),
+		}}
+	}
+	if accountJSON, accErrs := fetcher.FetchAccount(ctx, accountID); len(accErrs) > 0 || accountJSON == nil {
+		// accountID does not reference a valid account
+		if len(accErrs) > 0 {
+			errs = append(errs, errs...)
```
Comment (SyntaxNode) [inline account/account.go:31]: Caught this while reviewing. Shouldn't it be `errs = append(errs, accErrs...)`? .. and, you don't even need the `len(accErrs) > 0` check. `append` and the variadic operator handles null just fine.

**Comment by bsardo** [comment]
Looks good except for @SyntaxNode's catch.

**Comment by laurb9** [comment]
Updated to add back eventual fetcher errors.

File: account/account.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	if accountJSON, accErrs := fetcher.FetchAccount(ctx, accountID); len(accErrs) > 0 || accountJSON == nil {
+		// accountID does not reference a valid account
+		for _, e := range accErrs {
+			if _, ok := e.(stored_requests.NotFoundError); !ok {
+				errs = append(errs, e)
+			}
+		}
+		if cfg.AccountRequired && cfg.AccountDefaults.Disabled {
+			errs = append(errs, &errortypes.AcctRequired{
```
Comment (SyntaxNode) [inline account/account.go:36]: If there are `FetchAccount` errors other than `NotFoundError`, then the error message here implies the publisher is sending an invalid account id where the host may be experiencing problems accessing it's accounts store. How do you feel about adding a differently worded message here when `accErrs` > 0? Perhaps something alongs the lines of:

Prebid-server could not verify the Account ID. Please reach out to the prebid server host.

Comment (laurb9) [reply account/account.go:36]: Sounds better. Best to avoid misleading messaging.
updated in cc4a174

**Review by SyntaxNode — APPROVED** [review]

**Review by bsardo — APPROVED** [review]

**Pull request merged**